File: backend/models/vote.go
package models

type Vote struct {
	ID         int64      `json:"id"`
	TeamNumber TeamNumber `json:"team_number"`
	Type       VoteType   `json:"type"`
	MapName    string     `json:"map_name"`
	VetoID     int64      `json:"-"`
}

const (
	Ban    VoteType = "ban"
	Pick   VoteType = "pick"
	Random VoteType = "random"
)

type VoteType string

const (
	TeamOne TeamNumber = 1
	TeamTwo TeamNumber = 2
	Server  TeamNumber = 3
)

type TeamNumber int

File: src/renderer/types/vote.ts
import { VoteType } from 'renderer/types/vote-type';
import { TeamNumber } from 'renderer/types/team-number';
import { VoteStatus } from 'renderer/types/vote-status';

export type Vote = {
  id: number;
  type: VoteType;
  teamNumber: TeamNumber;
  status?: VoteStatus;
  mapName?: string;
};
